2009-02-13  Michael Tautschnig  <mt@debian.org>

	* examples/simple/files/etc/ucf.conf/FAIBASE: Added example ucf configuration
		that enforces choosing the default option non-interactively (closes #313397)
	* examples/simple/files/etc/apt/apt.conf.d/force_confdef/FAIBASE: Added dpkg
		option --force-confdef for apt/aptitude to make it non-interactive when
		config files have been changed (closes: #502394)
	* examples/simple/hooks/instsoft.FAIBASE: Copy those files before installing
		any packages
Index: trunk/examples/simple/files/etc/ucf.conf/FAIBASE
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ trunk/examples/simple/files/etc/ucf.conf/FAIBASE	
@@ -0,0 +1,39 @@
+#
+# This file is a bourne shell snippet, and is sourced by the
+# ucf script for configuration.
+#
+
+# Debugging information: The default value is 0 (no debugging
+# information is printed). To change the default behavior, uncomment
+# the following line and set the value to 1.
+#
+# DEBUG=0
+
+# Verbosity: The default value is 0 (quiet). To change the default
+# behavior, uncomment the following line and set the value to 1.
+#
+# VERBOSE=0
+
+
+# The src directory. This is the directory where the historical
+# md5sums for a file are looked for.  Specifically, the historical
+# md5sums are looked for in the subdirectory ${filename}.md5sum.d/
+#
+# conf_source_dir=/some/path/
+
+# Force the installed file to be retained. The default is have this
+# variable unset, which makes the script ask in case of doubt. To
+# change the default behavior, uncomment the following line and set
+# the value to YES
+#
+conf_force_conffold=YES
+
+# Force the installed file to be overridden. The default is have this
+# variable unset, which makes the script ask in case of doubt. To
+# change the default behavior, uncomment the following line and set
+# the value to YES
+#
+# conf_force_conffnew=YES
+
+# Please note that only one of conf_force_conffold and
+# conf_force_conffnew should be set.
Index: trunk/examples/simple/files/etc/apt/apt.conf.d/force_confdef/FAIBASE
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ trunk/examples/simple/files/etc/apt/apt.conf.d/force_confdef/FAIBASE	
@@ -0,0 +1,7 @@
+DPkg
+{
+  Options {
+	  "--force-confdef";
+	  }
+};
+
Index: trunk/examples/simple/hooks/instsoft.FAIBASE
===================================================================
--- trunk.orig/examples/simple/hooks/instsoft.FAIBASE
+++ trunk/examples/simple/hooks/instsoft.FAIBASE	
@@ -1,3 +1,6 @@
 #! /bin/bash
 
 [ -s $target/etc/kernel-img.conf ] || fcopy -Bi /etc/kernel-img.conf
+fcopy -Bi /etc/ucf.conf
+fcopy -Bi /etc/apt/apt.conf.d/force_confdef
+
